### PR TITLE
Update documentation by removing broken links and obsolete locations

### PR DESCRIPTION
**Pull request by Vigneswaran-Chandrasekaran**
Checked the documentation and updated out-of-date items, however I felt everything is updated on sections other than `Introduction` and `Developer's Guide` as other sections are mostly docstring generated. 
Kindly correct me if anything is incorrect. Please mention any particular sections which I would have possibly missed out and likely to need update.

### CODE REVIEW

File: docs_sphinx/developer/guidelines/documentation.rst [hunk 1]
```diff
@@ -4,9 +4,7 @@ Documentation
 It is very important to maintain documentation. We use the
 `Sphinx documentation generator <http://www.sphinx-doc.org/en/stable/>`__
 tools. The documentation is all hand written. Sphinx source files are stored in the
-``docs_sphinx`` folder (currently: ``dev/brian2/docs_sphinx``). The HTML files
-can be generated via the script ``dev/tools/docs/build_html_brian2.py`` and end
-up in the ``docs`` folder (currently: ``dev/brian2/docs``).
+``docs_sphinx`` folder.
```
Comment (mstimberg) [inline docs_sphinx/developer/guidelines/documentation.rst]: The change is correct, but it should keep the references to the ``dev/tools/docs/build_html_brian2.py`` script and the `docs`` folder.

File: docs_sphinx/developer/guidelines/documentation.rst [hunk 2]
```diff
@@ -47,11 +45,11 @@ is obvious what it does. For example, there is normally no need to document
 ``__str__`` with "Return a string representation.".
 
 For the docstring format, we use the our own sphinx extension (in
-`brian2.utils.sphinxext`) based on
+`brian2/sphinxext`) based on
```
Comment (mstimberg) [inline docs_sphinx/developer/guidelines/documentation.rst]: Please remove the trailing whitespace in this line

File: docs_sphinx/introduction/install.rst [hunk 3]
```diff
@@ -50,8 +48,8 @@ Installing Brian 2
     :ref:`installation_from_source` instead.
 
 You can either install Brian 2 in the Anaconda root environment, or create a
-new environment for Brian 2 (http://conda.pydata.org/docs/using/envs.html). The
-latter has the advantage that you can update (or not update) the dependencies
+new environment for Brian 2 (https://docs.anaconda.com/anaconda/navigator/tutorials/manage-environments/).
```
Comment (mstimberg) [inline docs_sphinx/introduction/install.rst]: Maybe rather use ``https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html``? We never talk about the Anaconda Navigator elsewhere.

File: docs_sphinx/developer/guidelines/documentation.rst [hunk 4]
```diff
@@ -4,7 +4,9 @@ Documentation
 It is very important to maintain documentation. We use the
 `Sphinx documentation generator <http://www.sphinx-doc.org/en/stable/>`__
 tools. The documentation is all hand written. Sphinx source files are stored in the
-``docs_sphinx`` folder.                                                      
+``docs_sphinx`` folder. The HTML files can be generated via the script
+``dev/tools/docs/build_html_brian2.py`` and end
+up in the ``docs`` folder (currently: ``dev/tools/docs``).
```
Comment (mstimberg) [inline docs_sphinx/developer/guidelines/documentation.rst]: We are almost there :) Can you remove the `(currently...)` part -- the documentation ends up in the `docs` folder directly in the main directory (i.e. parallel to ``docs_sphinx``), no need to add anything else.

Comment (Vigneswaran-Chandrasekaran) [reply docs_sphinx/developer/guidelines/documentation.rst]: Oh!! that was really a myopic change... Sorry for that...

**Pull request merged**